tests/testthat/test-FunctionReporter.R

```diff
@@ -156,6 +156,57 @@ test_that("FunctionReporter rejects bad packages with an informative error", {
     }, regexp = "pkgnet could not find a package called 'w0uldNEverB33aPackageName'")
 })
 
+### NETWORK EXTRACTION HELPER FUNCTIONS
+
+test_that(".parse_function correctly parses expressions for symbols", {
+    # Correctly parses body of function and finds all function symbols
+    expect_true({
+        myfunc <- function() {
+            x <- innerfunc1()
+            y <- innerfunc2()
+            z <- innerfunc3(innerfunc4())
+            2+2
+        }
+        result <- pkgnet:::.parse_function(body(myfunc))
+        all(c("innerfunc1", "innerfunc2", "innerfunc3", "innerfunc4") %in% result)
+    })
+})
+
+test_that(".parse_function correctly ignores right side of list extraction", {
+    # Correctly keeps left side of $ but drops right side of $
+    expect_true({
```

Comment:
does this also need an `expect_false` checking that `$listitem` was excluded

Reply:
My test condition captures both: `"myfunc" %in% result & !("listitem" %in% result)`